### DISCUSSION TITLE
Keep Source and Test module names in sync

### DISCUSSION THREAD

**Discussion by grrinchas** [jfmengels/elm-review#101] [Rule ideas]
**What the rule should do:**

Report Test modules which does not have corresponding Source module.

**What problems does it solve:**

I had this problem when you create a module, then write tests for that module, and then you forget about that module. Later you decide to move the module into different folder or rename it. But you forget to do the same for the corresponding Test module. So your Test modules and Source modules go out of sync. Therefore, 

* Can't figure out which Source module the Test module is testing
* Is hard to find the the Test module for particular Source

**Example of things the rule would report:**

```elm
User.elm
```
```elm
Public.UserTest.elm
``` 

In this case, rule would expect `Public.User.elm` to be present. 

**Example of things the rule would not report:**

```elm
User.elm
```
```elm
UserTest.elm
``` 

**Trivia:**

In most cases when you create a module you have the corresponding test module with the same name but with `Test` suffix. This rule enforces the standard.

****NOTE!** This is not about making sure that every Source module must have the Test module. But is that if you do have, then make sure that Test module name is consistent.**

**When (not) to enable this rule:**

I see no reason why it should be disabled. Besides, having this rule, would allow us to write other rules which depends on this one.

**I am looking for:**

I have already written this rule and been using for a while. Wondering if I should publish it or not? And if publish, whether under my own name or more official one?

**Comment by sparksp** [comment]
I just wanted to throw in another layout which I have adopted in my projects, I have a `Tests` module in my `tests` directory where everything sits under. For example:

- `Ui.Card` tested by `Tests.Ui.Card`

So I would be in favour of this rule with suffix vs prefix flavours.
